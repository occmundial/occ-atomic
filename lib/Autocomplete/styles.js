import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
