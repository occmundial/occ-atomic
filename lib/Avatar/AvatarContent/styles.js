import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
