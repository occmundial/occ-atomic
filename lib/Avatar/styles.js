import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
