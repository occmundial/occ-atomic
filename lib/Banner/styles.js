import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
