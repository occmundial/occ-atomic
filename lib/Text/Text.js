import React, { Component } from 'react';
import PropTypes from 'prop-types';

const oldToNewClassMapper = {
    // Text
    hero: 'h1',
    headline: 'h2',
    heading: 'h4',
    subheading: 'h5',
    extraLarge: 'bodyXLarge',
    large: 'bodyLarge',
    standard: 'bodyRegular',
    small: 'bodySmall',
    micro: 'bodyXSmall',
    // Color
    primary: 'indigoPrimary',
    secondary: 'pinkPrimary',
    white: 'whitePrimary',
    highEmphasis: 'corpPrimary',
    midEmphasis: 'corpSecondary',
    lowEmphasis: 'corpDisabled',
    // Spacing
    topXTiny: 'top1',
    topTiny: 'top2',
    topSmall: 'top4',
    topBase: 'top5',
    topMedium: 'top6',
    topLarge: 'top8',
    topXLarge: 'top9',
    bottomXTiny: 'bottom1',
    bottomTiny: 'bottom2',
    bottomSmall: 'bottom4',
    bottomBase: 'bottom5',
    bottomMedium: 'bottom6',
    bottomLarge: 'bottom8',
    bottomXLarge: 'bottom9'
};

const getActiveKey = (array) => {
    for (const item of array) {
        const key = Object.keys(item)[0];
        if (item[key]) return key;
    }
};

class Text extends Component {
    getSize() {
        const {
            classes, hero, headline, heading, subheading, extraLarge, large, standard, small, micro, display,
            h1, h2, h3, h4, h5, tagText, bodyXLarge, bodyLargeStrong, bodyLarge, bodyRegularStrong, bodyRegular,
            bodySmallStrong, bodySmall, bodyXSmall
        } = this.props;
        const sizeActive = getActiveKey([
            { display }, { h1 }, { h2 }, { h3 }, { h4 }, { h5 }, { tag: tagText }, { bodyXLarge },
            { bodyLargeStrong }, { bodyLarge }, { bodyRegularStrong }, { bodyRegular }, { bodySmallStrong },
            { bodySmall }, { bodyXSmall }, { hero }, { headline }, { heading }, { subheading }, { extraLarge },
            { large }, { standard }, { small }, { micro }
        ]);
        if (!sizeActive) return classes.bodyRegular;
        const activeSizeParsed = oldToNewClassMapper[sizeActive];
        return classes[activeSizeParsed || sizeActive];
    }

    getEmphasis() {
        const { classes, mid, low } = this.props;
        if (low) return classes.corpDisabled;
        else if (mid) return classes.corpSecondary;
        else return classes.corpPrimary;
    }

    getColor() {
        const {
            classes, primary, secondary, success, error, warning, info, disabled, white, link,
            indigoPrimary, indigoSecondary, pinkPrimary, whiteSecondary, whitePrimary
        } = this.props;
        const colorActive = getActiveKey([
            { primary }, { secondary }, { success }, { error }, { warning }, { info }, { disabled }, { white }, { link },
            { indigoPrimary }, { indigoSecondary }, { pinkPrimary }, { whiteSecondary }, { whitePrimary },
        ]);
        if (!colorActive) return false;
        const colorActiveParsed = oldToNewClassMapper[colorActive];
        return classes[colorActiveParsed || colorActive];
    }

    getAlign() {
        const { classes, left, center, right } = this.props;
        if (left) return classes.left;
        else if (center) return classes.center;
        else if (right) return classes.right;
    }

    getTopSpacing() {
        const {
            classes, topXTiny, topTiny, topSmall, topBase, topMedium, topLarge, topXLarge,
            top0, top1, top2, top3, top4, top5, top6, top7, top8, top9, top10, top11, top12
        } = this.props;
        const spacing = [
            { topXTiny }, { topTiny }, { topSmall }, { topBase }, { topMedium }, { topLarge }, { topXLarge },
            { top0 }, { top1 }, { top2 }, { top3 }, { top4 }, { top5 }, { top6 }, { top7 }, { top8 },
            { top9 }, { top10 }, { top11 }, { top12 }
        ];
        const topSpacingActive = getActiveKey(spacing);
        if (!topSpacingActive) return '';
        const topSpacingActiveParsed = oldToNewClassMapper[topSpacingActive];
        return ` ${classes[topSpacingActiveParsed || topSpacingActive]}`;
    }

    getBottomSpacing() {
        const { classes, bottomXTiny, bottomTiny, bottomSmall, bottomBase, bottomMedium, bottomLarge, bottomXLarge,
            bottom0, bottom1, bottom2, bottom3, bottom4, bottom5, bottom6, bottom7, bottom8, bottom9, bottom10, bottom11, bottom12,
         } = this.props;
        const spacing = [{ bottomXTiny }, { bottomTiny }, { bottomSmall }, { bottomBase }, { bottomMedium }, { bottomLarge }, { bottomXLarge },
            { bottom0 }, { bottom1 }, { bottom2 }, { bottom3 }, { bottom4 }, { bottom5 }, { bottom6 }, { bottom7 }, { bottom8 }, { bottom9 }, 
            { bottom10 }, { bottom11 }, { bottom12 }
        ];
        const bottomSpacingActive = getActiveKey(spacing);
        if (!bottomSpacingActive) return '';
        const bottomSpacingActiveParsed = oldToNewClassMapper[bottomSpacingActive];
        return ` ${classes[bottomSpacingActiveParsed || bottomSpacingActive]}`;
    }

    getWeight() {
        const { classes, hero, headline, heading, subheading, large, standard, small, micro, strong } = this.props;
        const sizes = [{ hero }, { headline }, { heading }, { subheading }, { large }, { standard }, { small }, { micro }];
        const weightActive = getActiveKey(sizes);
        if ((!weightActive || large || standard || small || micro) && strong)
            return ` ${classes.strong}`;
        return false;
    }

    render() {
        const { classes, children, tag, className, style, id, testId } = this.props;
        const TextType = tag;
        const size = this.getSize();
        const color = this.getColor();
        const emphasis = this.getEmphasis();
        const align = this.getAlign();
        const topSpacing = this.getTopSpacing();
        const bottomSpacing = this.getBottomSpacing();
        const strong = this.getWeight();
        return (
            <TextType
                className={`${classes.text} ${size} ${color ? color : emphasis}${topSpacing}${bottomSpacing}${align ? ` ${align}` : ''}${strong ? ` ${strong}` : ''}${className ? ` ${className}` : ''}`}
                style={style}
                id={id}
                data-testid={testId}
            >
                {children}
            </TextType>
        );
    }
}

Text.propTypes = {
    /** display typography style. */
    display: PropTypes.bool,
    /** h1 typography style. */
    h1: PropTypes.bool,
    /** h2 typography style. */
    h2: PropTypes.bool,
    /** h3 typography style. */
    h3: PropTypes.bool,
    /** h4 typography style. */
    h4: PropTypes.bool,
    /** h5 typography style. */
    h5: PropTypes.bool,
    /** tagText typography style. */
    tagText: PropTypes.bool,
    /** bodyXLarge typography style. */
    bodyXLarge: PropTypes.bool,
    /** bodyLargeStrong typography style. */
    bodyLargeStrong: PropTypes.bool,
    /** bodyLarge typography style. */
    bodyLarge: PropTypes.bool,
    /** bodyRegularStrong typography style. */
    bodyRegularStrong: PropTypes.bool,
    /** bodyRegular typography style. */
    bodyRegular: PropTypes.bool,
    /** bodySmallStrong typography style. */
    bodySmallStrong: PropTypes.bool,
    /** bodySmall typography style. */
    bodySmall: PropTypes.bool,
    /** bodyXSmall typography style. */
    bodyXSmall: PropTypes.bool,
    /** Indigo primary emphasis color style. */
    indigoPrimary: PropTypes.bool,
    /** Indigo secondary emphasis color style. */
    indigoSecondary: PropTypes.bool,
    /** Pink primary emphasis color style. */
    pinkPrimary: PropTypes.bool,
    /** White secondary  emphasis color style. */
    whiteSecondary: PropTypes.bool,
    /** White  emphasis color style. */
    whitePrimary: PropTypes.bool,
    /** Corp secondary emphasis color style. */
    corpSecondary: PropTypes.bool,
    /** Corp disabled emphasis color style. */
    corpDisabled: PropTypes.bool,
    /** Top0 spacing style (2px). */
    top0: PropTypes.bool,
    /** Top1 spacing style (4px). */
    top1: PropTypes.bool,
    /** Top2 spacing style (8px). */
    top2: PropTypes.bool,
    /** Top3 spacing style (12px). */
    top3: PropTypes.bool,
    /** Top4 spacing style (16px). */
    top4: PropTypes.bool,
    /** Top5 spacing style (24px). */
    top5: PropTypes.bool,
    /** Top6 spacing style (32px). */
    top6: PropTypes.bool,
    /** Top7 spacing style (40px). */
    top7: PropTypes.bool,
    /** Top8 spacing style (48px). */
    top8: PropTypes.bool,
    /** Top9 spacing style (64px). */
    top9: PropTypes.bool,
    /** Top10 spacing style (80px). */
    top10: PropTypes.bool,
    /** Top11 spacing style (120px). */
    top11: PropTypes.bool,
    /** Top12 spacing style (160px). */
    top12: PropTypes.bool,
    /** Bottom0 spacing style (2px). */
    bottom0: PropTypes.bool,
    /** Bottom1 spacing style (4px). */
    bottom1: PropTypes.bool,
    /** Bottom2 spacing style (8px). */
    bottom2: PropTypes.bool,
    /** Bottom3 spacing style (12px). */
    bottom3: PropTypes.bool,
    /** Bottom4 spacing style (16px). */
    bottom4: PropTypes.bool,
    /** Bottom5 spacing style (24px). */
    bottom5: PropTypes.bool,
    /** Bottom6 spacing style (32px). */
    bottom6: PropTypes.bool,
    /** Bottom7 spacing style (40px). */
    bottom7: PropTypes.bool,
    /** Bottom8 spacing style (48px). */
    bottom8: PropTypes.bool,
    /** Bottom9 spacing style (64px). */
    bottom9: PropTypes.bool,
    /** Bottom10 spacing style (80px). */
    bottom10: PropTypes.bool,
    /** Bottom11 spacing style (120px). */
    bottom11: PropTypes.bool,
    /** Bottom12 spacing style (160px). */
    bottom12: PropTypes.bool,

    classes: PropTypes.object.isRequired,
    children: PropTypes.node.isRequired,
    /** Strong style. Only applies in large, standard or small sizes. */
    strong: PropTypes.bool,
    /** Success color style. */
    success: PropTypes.bool,
    /** Error color style. */
    error: PropTypes.bool,
    /** Warning color style. */
    warning: PropTypes.bool,
    /** Left align style. */
    left: PropTypes.bool,
    /** Center align style. */
    center: PropTypes.bool,
    /** Right align style. */
    right: PropTypes.bool,
    /** Html tag. */
    tag: PropTypes.string.isRequired,
    /** The className of the text container. */
    className: PropTypes.string,
    /** The style of the text container. */
    style: PropTypes.object,
    id: PropTypes.string,
    testId: PropTypes.string,

    /** [Legacy prop] h1 typography style.*/
    hero: PropTypes.bool,
    /** [Legacy prop] h2 typography style.*/
    headline: PropTypes.bool,
    /** [Legacy prop] h4 typography style.*/
    heading: PropTypes.bool,
    /** [Legacy prop] h5 typography style.*/
    subheading: PropTypes.bool,
    /** [Legacy prop] bodyXLarge typography style.*/
    extraLarge: PropTypes.bool,
    /** [Legacy prop] bodyLarge typography style.*/
    large: PropTypes.bool,
    /** [Legacy prop] bodyRegular typography style.*/
    standard: PropTypes.bool,
    /** [Legacy prop] bodySmall typography style.*/
    small: PropTypes.bool,
    /** [Legacy prop] bodyXSmall typography style.*/
    micro: PropTypes.bool,
    /** [Legacy prop] Corp secondary emphasis color style.*/
    mid: PropTypes.bool,
    /** [Legacy prop] Corp disabled emphasis color style.*/
    low: PropTypes.bool,
    /** [Legacy prop] Indigo primary emphasis color style.*/
    primary: PropTypes.bool,
    /** [Legacy prop] Pink primary emphasis color style.*/
    secondary: PropTypes.bool,
    /** [Legacy prop] Info color style.*/
    info: PropTypes.bool,
    /** [Legacy prop] Disabled emphasis color style.*/
    disabled: PropTypes.bool,
    /** [Legacy prop] White  emphasis color style.*/
    white: PropTypes.bool,
    /** [Legacy prop] Link emphasis color style.*/
    link: PropTypes.bool,
    /** [Legacy prop] Top1 spacing style (4px).*/
    topXTiny: PropTypes.bool,
    /** [Legacy prop] Top2 spacing style (8px).*/
    topTiny: PropTypes.bool,
    /** [Legacy prop] Top4 spacing style (16px).*/
    topSmall: PropTypes.bool,
    /** [Legacy prop] Top5 spacing style (24px).*/
    topBase: PropTypes.bool,
    /** [Legacy prop] Top6 spacing style (32px).*/
    topMedium: PropTypes.bool,
    /** [Legacy prop] Top8 spacing style (48px).*/
    topLarge: PropTypes.bool,
    /** [Legacy prop] Top9 spacing style (64px).*/
    topXLarge: PropTypes.bool,
    /** [Legacy prop] Bottom1 spacing style (4px).*/
    bottomXTiny: PropTypes.bool,
    /** [Legacy prop] Bottom2 spacing style (8px).*/
    bottomTiny: PropTypes.bool,
    /** [Legacy prop] Bottom4 spacing style (16px).*/
    bottomSmall: PropTypes.bool,
    /** [Legacy prop] Bottom5 spacing style (24px).*/
    bottomBase: PropTypes.bool,
    /** [Legacy prop] Bottom6 spacing style (32px).*/
    bottomMedium: PropTypes.bool,
    /** [Legacy prop] Bottom8 spacing style (48px).*/
    bottomLarge: PropTypes.bool,
    /** [Legacy prop] Bottom9 spacing style (64px).*/
    bottomXLarge: PropTypes.bool
};

Text.defaultProps = {
    tag: 'p'
};

export default Text;
