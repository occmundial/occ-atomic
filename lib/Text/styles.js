import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import grid from '../subatomic/grid';
import spacing from '../subatomic/spacing';

export default {
    text: {
        letterSpacing:0,
        fontFamily:fonts.body,
        fontWeight:400,
        fontStyle:'normal',
        margin:0
    },
    // Sizes
    hero: {
        fontSize:28,
        lineHeight:'32px',
        fontWeight: 600,
        [`@media screen and (min-width:${grid.sm}px)`]: {
            fontSize:57,
            lineHeight:'72px',
            letterSpacing:-2
        }
    },
    headline: {
        fontSize:28,
        lineHeight:'32px',
        fontWeight: 600,
        [`@media screen and (min-width:${grid.sm}px)`]: {
            fontSize:36,
            lineHeight:'48px',
            letterSpacing:-0.5
        }
    },
    heading: {
        fontSize:22,
        lineHeight:'32px',
        fontWeight:600
    },
    subheading: {
        fontSize:17,
        lineHeight:'24px',
        fontWeight:600
    },
    extraLarge: {
        fontSize:22,
        lineHeight:'32px'
    },
    large: {
        fontSize:17,
        lineHeight:'24px'
    },
    standard: {
        fontSize:15,
        lineHeight:'24px'
    },
    small: {
        fontSize:13,
        lineHeight:'16px'
    },
    micro: {
        fontSize:11,
        lineHeight:'16px'
    },
    // Weight
    strong: {
        fontWeight: 600
    },
    // Colors
    primary: {
        color:colors.prim
    },
    secondary: {
        color:colors.sec
    },
    success: {
        color:colors.successText
    },
    error: {
        color:colors.errorText
    },
    warning: {
        color:colors.warningText
    },
    info: {
        color:colors.infoText
    },
    disabled: {
        color:colors.inkLightest
    },
    white: {
        color:colors.white
    },
    link: {
        color:colors.textLink
    },
    // Emphasis
    highEmphasis: {
        color:colors.ink
    },
    midEmphasis: {
        color:colors.inkLight
    },
    lowEmphasis: {
        color:colors.inkLighter
    },
    // Align
    left: {
        textAlign:'left'
    },
    center: {
        textAlign:'center'
    },
    right: {
        textAlign:'right'
    },
    // Spacing
    topXTiny: {
        marginTop:spacing.xTiny
    },
    topTiny: {
        marginTop:spacing.tiny
    },
    topSmall: {
        marginTop:spacing.small
    },
    topBase: {
        marginTop:spacing.base
    },
    topMedium: {
        marginTop:spacing.medium
    },
    topLarge: {
        marginTop:spacing.large
    },
    topXLarge: {
        marginTop:spacing.xLarge
    },
    bottomXTiny: {
        marginBottom:spacing.xTiny
    },
    bottomTiny: {
        marginBottom:spacing.tiny
    },
    bottomSmall: {
        marginBottom:spacing.small
    },
    bottomBase: {
        marginBottom:spacing.base
    },
    bottomMedium: {
        marginBottom:spacing.medium
    },
    bottomLarge: {
        marginBottom:spacing.large
    },
    bottomXLarge: {
        marginBottom:spacing.xLarge
    }
};
