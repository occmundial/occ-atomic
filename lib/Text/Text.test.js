import React from 'react';
import { shallow } from 'enzyme';
import Text from './Text';
import styles from './styles';
import TextJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Text", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Text classes={classes}>Text</Text>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    it('has the correct size style', () => {
        const wrapper = shallow(
            <Text hero classes={classes}>Text</Text>
        );
        expect(wrapper.find('.h1').length).toBe(1);
    });
    
    it('has a default color style', () => {
        const wrapper = shallow(
            <Text classes={classes}>Text</Text>
        );
        expect(wrapper.find('.corpPrimary').length).toBe(1);
    });
    
    it('has the correct color style', () => {
        const wrapper = shallow(
            <Text primary classes={classes}>Text</Text>
        );
        expect(wrapper.find('.indigoPrimary').length).toBe(1);
    });
    
    it('has the correct align style', () => {
        const wrapper = shallow(
            <Text center classes={classes}>Text</Text>
        );
        expect(wrapper.find('.center').length).toBe(1);
    });
    
    it('has the correct spacing style', () => {
        const wrapper = shallow(
            <Text topTiny bottomLarge classes={classes}>Text</Text>
        );
        expect(wrapper.find('.top2').length).toBe(1);
        expect(wrapper.find('.bottom8').length).toBe(1);
    });

    it('has the correct html tag', () => {
        const wrapper = shallow(
            <Text tag="h3" classes={classes}>Text</Text>
        );
        expect(wrapper.find('h3').length).toBe(1);
    });

});

describe("TextJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <TextJSS>Text</TextJSS>
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Text styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});