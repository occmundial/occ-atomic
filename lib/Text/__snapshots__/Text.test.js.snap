// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    classes={
      Object {
        "bottomBase": "bottomBase",
        "bottomLarge": "bottomLarge",
        "bottomMedium": "bottomMedium",
        "bottomSmall": "bottomSmall",
        "bottomTiny": "bottomTiny",
        "bottomXLarge": "bottomXLarge",
        "bottomXTiny": "bottomXTiny",
        "center": "center",
        "disabled": "disabled",
        "error": "error",
        "heading": "heading",
        "headline": "headline",
        "hero": "hero",
        "highEmphasis": "highEmphasis",
        "info": "info",
        "large": "large",
        "left": "left",
        "link": "link",
        "lowEmphasis": "lowEmphasis",
        "micro": "micro",
        "midEmphasis": "midEmphasis",
        "primary": "primary",
        "right": "right",
        "secondary": "secondary",
        "small": "small",
        "standard": "standard",
        "strong": "strong",
        "subheading": "subheading",
        "success": "success",
        "text": "text",
        "topBase": "topBase",
        "topLarge": "topLarge",
        "topMedium": "topMedium",
        "topSmall": "topSmall",
        "topTiny": "topTiny",
        "topXLarge": "topXLarge",
        "topXTiny": "topXTiny",
        "warning": "warning",
        "white": "white",
      }
    }
    tag="p"
  >
    Text
  </Text>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Text",
      "className": "text standard highEmphasis",
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": "Text",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Text",
        "className": "text standard highEmphasis",
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": "Text",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Text styles matches the snapshot 1`] = `
Object {
  "bottomBase": Object {
    "marginBottom": 24,
  },
  "bottomLarge": Object {
    "marginBottom": 48,
  },
  "bottomMedium": Object {
    "marginBottom": 32,
  },
  "bottomSmall": Object {
    "marginBottom": 16,
  },
  "bottomTiny": Object {
    "marginBottom": 8,
  },
  "bottomXLarge": Object {
    "marginBottom": 64,
  },
  "bottomXTiny": Object {
    "marginBottom": 4,
  },
  "center": Object {
    "textAlign": "center",
  },
  "disabled": Object {
    "color": "#dddddd",
  },
  "error": Object {
    "color": "#db3737",
  },
  "heading": Object {
    "fontSize": 22,
    "fontWeight": 600,
    "lineHeight": "32px",
  },
  "headline": Object {
    "@media screen and (min-width:768px)": Object {
      "fontSize": 36,
      "lineHeight": "48px",
    },
    "fontSize": 28,
    "lineHeight": "32px",
  },
  "hero": Object {
    "@media screen and (min-width:768px)": Object {
      "fontSize": 57,
      "lineHeight": "72px",
    },
    "fontSize": 28,
    "lineHeight": "32px",
  },
  "highEmphasis": Object {
    "color": "#222222",
  },
  "info": Object {
    "color": "#5736ab",
  },
  "large": Object {
    "fontSize": 16,
    "lineHeight": "24px",
  },
  "left": Object {
    "textAlign": "left",
  },
  "link": Object {
    "color": "#0946CB",
  },
  "lowEmphasis": Object {
    "color": "#aaaaaa",
  },
  "micro": Object {
    "fontSize": 10,
    "lineHeight": "16px",
  },
  "midEmphasis": Object {
    "color": "#777777",
  },
  "primary": Object {
    "color": "#083cae",
  },
  "right": Object {
    "textAlign": "right",
  },
  "secondary": Object {
    "color": "#f13465",
  },
  "small": Object {
    "fontSize": 12,
    "lineHeight": "16px",
  },
  "standard": Object {
    "fontSize": 14,
    "lineHeight": "24px",
  },
  "strong": Object {
    "fontWeight": 600,
  },
  "subheading": Object {
    "fontSize": 16,
    "fontWeight": 600,
    "lineHeight": "24px",
  },
  "success": Object {
    "color": "#009537",
  },
  "text": Object {
    "fontFamily": "'hk_grotesk', sans-serif",
    "fontStyle": "normal",
    "fontWeight": 400,
    "letterSpacing": 0,
    "margin": 0,
  },
  "topBase": Object {
    "marginTop": 24,
  },
  "topLarge": Object {
    "marginTop": 48,
  },
  "topMedium": Object {
    "marginTop": 32,
  },
  "topSmall": Object {
    "marginTop": 16,
  },
  "topTiny": Object {
    "marginTop": 8,
  },
  "topXLarge": Object {
    "marginTop": 64,
  },
  "topXTiny": Object {
    "marginTop": 4,
  },
  "warning": Object {
    "color": "#ffc549",
  },
  "white": Object {
    "color": "#ffffff",
  },
}
`;

exports[`TextJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Text)
    tag="p"
  >
    Text
  </Jss(Text)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Text",
      "classes": Object {
        "bottomBase": "Text-bottomBase-0-1-36",
        "bottomLarge": "Text-bottomLarge-0-1-38",
        "bottomMedium": "Text-bottomMedium-0-1-37",
        "bottomSmall": "Text-bottomSmall-0-1-35",
        "bottomTiny": "Text-bottomTiny-0-1-34",
        "bottomXLarge": "Text-bottomXLarge-0-1-39",
        "bottomXTiny": "Text-bottomXTiny-0-1-33",
        "center": "Text-center-0-1-24",
        "disabled": "Text-disabled-0-1-17",
        "error": "Text-error-0-1-14",
        "heading": "Text-heading-0-1-4",
        "headline": "Text-headline-0-1-3",
        "hero": "Text-hero-0-1-2",
        "highEmphasis": "Text-highEmphasis-0-1-20",
        "info": "Text-info-0-1-16",
        "large": "Text-large-0-1-6",
        "left": "Text-left-0-1-23",
        "link": "Text-link-0-1-19",
        "lowEmphasis": "Text-lowEmphasis-0-1-22",
        "micro": "Text-micro-0-1-9",
        "midEmphasis": "Text-midEmphasis-0-1-21",
        "primary": "Text-primary-0-1-11",
        "right": "Text-right-0-1-25",
        "secondary": "Text-secondary-0-1-12",
        "small": "Text-small-0-1-8",
        "standard": "Text-standard-0-1-7",
        "strong": "Text-strong-0-1-10",
        "subheading": "Text-subheading-0-1-5",
        "success": "Text-success-0-1-13",
        "text": "Text-text-0-1-1",
        "topBase": "Text-topBase-0-1-29",
        "topLarge": "Text-topLarge-0-1-31",
        "topMedium": "Text-topMedium-0-1-30",
        "topSmall": "Text-topSmall-0-1-28",
        "topTiny": "Text-topTiny-0-1-27",
        "topXLarge": "Text-topXLarge-0-1-32",
        "topXTiny": "Text-topXTiny-0-1-26",
        "warning": "Text-warning-0-1-15",
        "white": "Text-white-0-1-18",
      },
      "tag": "p",
    },
    "ref": null,
    "rendered": "Text",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Text",
        "classes": Object {
          "bottomBase": "Text-bottomBase-0-1-36",
          "bottomLarge": "Text-bottomLarge-0-1-38",
          "bottomMedium": "Text-bottomMedium-0-1-37",
          "bottomSmall": "Text-bottomSmall-0-1-35",
          "bottomTiny": "Text-bottomTiny-0-1-34",
          "bottomXLarge": "Text-bottomXLarge-0-1-39",
          "bottomXTiny": "Text-bottomXTiny-0-1-33",
          "center": "Text-center-0-1-24",
          "disabled": "Text-disabled-0-1-17",
          "error": "Text-error-0-1-14",
          "heading": "Text-heading-0-1-4",
          "headline": "Text-headline-0-1-3",
          "hero": "Text-hero-0-1-2",
          "highEmphasis": "Text-highEmphasis-0-1-20",
          "info": "Text-info-0-1-16",
          "large": "Text-large-0-1-6",
          "left": "Text-left-0-1-23",
          "link": "Text-link-0-1-19",
          "lowEmphasis": "Text-lowEmphasis-0-1-22",
          "micro": "Text-micro-0-1-9",
          "midEmphasis": "Text-midEmphasis-0-1-21",
          "primary": "Text-primary-0-1-11",
          "right": "Text-right-0-1-25",
          "secondary": "Text-secondary-0-1-12",
          "small": "Text-small-0-1-8",
          "standard": "Text-standard-0-1-7",
          "strong": "Text-strong-0-1-10",
          "subheading": "Text-subheading-0-1-5",
          "success": "Text-success-0-1-13",
          "text": "Text-text-0-1-1",
          "topBase": "Text-topBase-0-1-29",
          "topLarge": "Text-topLarge-0-1-31",
          "topMedium": "Text-topMedium-0-1-30",
          "topSmall": "Text-topSmall-0-1-28",
          "topTiny": "Text-topTiny-0-1-27",
          "topXLarge": "Text-topXLarge-0-1-32",
          "topXTiny": "Text-topXTiny-0-1-26",
          "warning": "Text-warning-0-1-15",
          "white": "Text-white-0-1-18",
        },
        "tag": "p",
      },
      "ref": null,
      "rendered": "Text",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
